tests/test_app.py

```diff
@@ -140,6 +162,127 @@ def test_app_init_with_invalid_name(self, name):
+                        config_json='firebase_config.json',
+                        init_options={},
+                        want_options={}),
+        OptionsTestCase(name='empty options are options, json is ignored',
+                        config_json='{"projectId": "hipster-chat-mock"}',
+                        init_options={},
+                        want_options={}),
+        OptionsTestCase(name='no options, partial config in file',
+                        config_json='firebase_config_partial.json',
+                        init_options=None,
+                        want_options={'databaseURL': 'https://hipster-chat.firebaseio.mock',
+                                      'projectId': 'hipster-chat-mock'}),
+        OptionsTestCase(name='no options, partial config in json',
+                        config_json='{"databaseURL": "https://hipster-chat.firebaseio.mock",' +
+                        '"projectId": "hipster-chat-mock"}',
+                        init_options=None,
+                        want_options={'databaseURL': 'https://hipster-chat.firebaseio.mock',
+                                      'projectId': 'hipster-chat-mock'}),
+        OptionsTestCase(name='partial config file is ignored',
+                        config_json='firebase_config_partial.json',
+                        init_options={'projectId': 'pid1-mock',
+                                      'storageBucket': 'sb1-mock'},
+                        want_options={'projectId': 'pid1-mock',
+                                      'storageBucket': 'sb1-mock'}),
+        OptionsTestCase(name='full config file is ignored',
+                        config_json='firebase_config.json',
+                        init_options={'databaseAuthVariableOverride': 'davy1-mock',
+                                      'databaseURL': 'https://db1-mock',
+                                      'projectId': 'pid1-mock',
+                                      'storageBucket': 'sb1-.mock'},
+                        want_options={'databaseAuthVariableOverride': 'davy1-mock',
+                                      'databaseURL': 'https://db1-mock',
+                                      'projectId': 'pid1-mock',
+                                      'storageBucket': 'sb1-.mock'}),
+        OptionsTestCase(name='full config file is ignored with missing values in options',
+                        config_json='firebase_config.json',
+                        init_options={'databaseAuthVariableOverride': 'davy1 - mock',
+                                      'projectId': 'pid1 - mock',
+                                      'storageBucket': 'sb1 - .mock'},
+                        want_options={'databaseAuthVariableOverride': 'davy1 - mock',
+                                      'projectId': 'pid1 - mock',
+                                      'storageBucket': 'sb1 - .mock'})]
+
+    @pytest.mark.parametrize('test_case', options_test_cases)
```

Comment:
Pass `ids=` parameter here. Then you don't have to use `test_case.name` in the body of the test case.